### ISSUE TITLE
Add DST/DCT Helmholtz and Poisson solvers for non-periodic BCs

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#26]
## Overview

spectraldiffx's Helmholtz solvers only support **periodic BCs** (via FFT). There is no support for:
- **Dirichlet BCs** (via DST) — zero-value boundaries
- **Neumann BCs** (via DCT) — zero-gradient boundaries

These are essential for bounded domain problems (channel flows, enclosed basins, no-slip walls).

## Missing Solvers

| Solver | BC Type | Transform | Notes |
|--------|---------|-----------|-------|
| `solve_poisson_dst(rhs, dx, dy)` | Dirichlet | DST Type-I | Bounded domains, no-slip walls |
| `solve_poisson_dct(rhs, dx, dy)` | Neumann | DCT Type-II | Bounded domains, free-slip/insulating |
| `solve_helmholtz_dst(rhs, λ, dx, dy)` | Dirichlet | DST Type-I | (∇² − λ)u = f, Dirichlet |
| `solve_helmholtz_dct(rhs, λ, dx, dy)` | Neumann | DCT Type-II | (∇² − λ)u = f, Neumann |

## Reference Implementation

**Source:** `jej_vc_snippets/linear_algebra/solver_spectral_jax.py`

The snippet provides:
- Eigenvalue computation: `dst_eigenvalues(N, dx)`, `dct_eigenvalues(N, dx)`, `fft_eigenvalues(N, dx)`
- DST types I-IV support (different grid configurations)
- DCT types I-IV support
- Nullspace handling for Poisson (λ=0): "raise", "pinv", or "zero_mean" policies
- Batch/multi-layer solving: shape `(*batch, ny, nx)`
- Proper normalization for each transform type

## Proposed API

```python
class SpectralHelmholtzSolverDST2D(eqx.Module):
    """Helmholtz solver with Dirichlet BCs via DST."""
    eigenvalues_x: Float[Array, "Nx"]
    eigenvalues_y: Float[Array, "Ny"]
    
    @classmethod
    def from_grid(cls, Nx, Ny, dx, dy): ...
    
    def solve(self, rhs: Float[Array, "*batch Ny Nx"], lam: float = 0.0) -> Float[Array, "*batch Ny Nx"]: ...

class SpectralHelmholtzSolverDCT2D(eqx.Module):
    """Helmholtz solver with Neumann BCs via DCT."""
    ...
```

## Note on finitevolX Overlap

finitevolX already has `solve_poisson_dst/dct/fft` and `solve_helmholtz_dst/dct/fft` in `_src/elliptic.py`. Once spectraldiffx has these, finitevolX could import from spectraldiffx instead.

## Target Module

`spectraldiffx/_src/solvers.py` (extend with DST/DCT solver classes)

## Acceptance Criteria

- [ ] DST-based Poisson solver (2D, with 1D and 3D variants)
- [ ] DCT-based Poisson solver (2D, with 1D and 3D variants)
- [ ] DST-based Helmholtz solver (2D, with 1D and 3D variants)
- [ ] DCT-based Helmholtz solver (2D, with 1D and 3D variants)
- [ ] Eigenvalue helper functions for DST/DCT/FFT
- [ ] Nullspace handling policy for Poisson (λ=0)
- [ ] Batch/multi-layer support `(*batch, ny, nx)`
- [ ] Tests against analytical solutions

## Priority

**HIGH** — non-periodic BCs are essential for bounded domain problems.
